### ISSUE TITLE
[BUG] taint-toleration default-settings not correctly merged with existing ones

### ISSUE THREAD

**Issue by Wykiki** [longhorn/longhorn#4335]
## Describe the bug

Upgrading from Longhorn 1.2.3 to 1.3.0, I encountered longhorn-manager deployment fail.
The logs are below

## To Reproduce

1. Set a value for taint-toleration in Chart values.
2. Install Longhorn via Rancher UI
3. Wait
4. Installation fails with errors on longhorn-manager pods
5. See error

## Expected behavior

Longhorn correctly upgrade, especially if nothing has been modified on the defaults-settings / values.yaml.

## Log or Support bundle

Logs of longhorn-manager :
```
time="2022-08-01T06:48:25Z" level=warning msg="Error syncing Longhorn volume longhorn-system/pvc-2ed0a2b1-9f4e-4add-a66b-ab1ef9e880f4" controller=longhorn-volume error="fail to sync longhorn-system/pvc-2ed0a2b1-9f4e-4add-a66b-ab1ef9e880f4: unable to get engine image rancher/mirrored-longhornio-longhorn-engine:v1.3.0: engineimage.longhorn.io \"ei-dae99989\" not found" node=storage-00
time="2022-08-01T06:48:25Z" level=warning msg="Error syncing ConfigMap longhorn-system/longhorn-default-setting" controller=longhorn-kubernetes-configmap-controller error="longhorn-kubernetes-configmap-controller: fail to sync longhorn-system/longhorn-default-setting: failed to update built-in settings with customized values: fail to set the setting taint-toleration with invalid value kubernetes.engie.com/node-type=storage:NoExecute: cannot modify toleration setting before all volumes are detached" node=storage-00
```

## Environment

 - Longhorn version: 1.3.0
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Rancher
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 6
 - Node config
   - OS type and version: Ubuntu 20.04
   - CPU per node: 16
   - Memory per node: 64
   - Disk type(e.g. SSD/NVMe): HDD 7200k
   - Network bandwidth between the nodes: ?
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster: 70

## Additional context

As workaround, I've set the taint-toleration as null, then Longhorn upgrades successfully, and I still have the correct taint-toleration from the configuration panel on the UI.

**Comment by derekbit** [comment]
@Wykiki 
Thanks for the information. This issue is fixed and merged in the upcoming v1.3.1.
Ref: https://github.com/longhorn/longhorn/issues/4160

**Comment by Wykiki** [comment]
Sorry, didn't give a look at existing issues, was in a rush :s
Anyway, thanks for your fast reply !
